test/e2e/tests/4byte-directory.spec.js

```diff
@@ -0,0 +1,110 @@
+const {
+  withFixtures,
+  openDapp,
+  unlockWallet,
+  largeDelayMs,
+  veryLargeDelayMs,
+  WINDOW_TITLES,
+} = require('../helpers');
+const { SMART_CONTRACTS } = require('../seeder/smart-contracts');
+
+describe('4byte setting', function () {
+  it('makes a call to 4byte when the setting is on', async function () {
+    const smartContract = SMART_CONTRACTS.PIGGYBANK;
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        smartContract,
+        title: this.test.title,
+      },
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
+        await driver.navigate();
+        await unlockWallet(driver);
+
+        // deploy contract
+        await openDapp(driver, contractAddress);
+
+        // wait for deployed contract, calls and confirms a contract method where ETH is sent
+        await driver.delay(largeDelayMs);
+        await driver.clickElement('#depositButton');
+
+        await driver.waitForSelector({
+          css: 'span',
+          text: 'Deposit initiated',
+        });
+
+        await driver.waitUntilXWindowHandles(3);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Notification);
+        const actionElement = await driver.waitForSelector({
+          css: '.confirm-page-container-summary__action__name',
+          text: 'Deposit',
+        });
+        assert.equal(await actionElement.getText(), 'DEPOSIT');
+      },
+    );
+  });
+
+  it('does not try to get contract method name from 4byte when the setting is off', async function () {
+    const smartContract = SMART_CONTRACTS.PIGGYBANK;
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        smartContract,
+        title: this.test.title,
+      },
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
+        await driver.navigate();
+        await unlockWallet(driver);
+
+        // goes to the settings screen
+        await driver.clickElement(
+          '[data-testid="account-options-menu-button"]',
+        );
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.clickElement({ text: 'Security & privacy', tag: 'div' });
+
+        // turns off 4Byte Directory contract method name resolution
+        await driver.clickElement(
+          '[data-testid="4byte-resolution-container"] .toggle-button',
+        );
+
+        // deploy contract
+        await openDapp(driver, contractAddress);
+
+        // wait for deployed contract, calls and confirms a contract method where ETH is sent
+        await driver.delay(largeDelayMs);
```

Comment:
see [comment below](https://github.com/MetaMask/metamask-extension/pull/20098/files#r1281598365)

Reply:
Can we use `findClickableElement` against the deposit button instead?